### PR TITLE
Polish documentation

### PR DESCRIPTION
**Pull request by nipafx**
Proposed commit message:

```
Polish documentation (#681, #685 / #690)

* improve project description
* improve article titles, descriptions, and first paragraph (~> #681)
* sort navigation by title
* fix broken internal links
* use `====` for multi-line admonition boxes
* many small improvements (~> #685)

Closes: #681, #685
PR: #690
```

---
**PR checklist**

The following checklist shall help the PR's author, the reviewers and maintainers to ensure the quality of this project.
It is based on our contributors guidelines, especially the ["writing code" section](https://github.com/junit-pioneer/junit-pioneer/blob/main/CONTRIBUTING.md#writing-code).
It shall help to check for completion of the listed points.
If a point does not apply to the given PR's changes, the corresponding entry can be simply marked as done. 

Documentation (general)
* [x] There is documentation (Javadoc and site documentation; added or updated)
* [x] There is implementation information to describe _why_ a non-obvious source code / solution got implemented
* [x] Site documentation has its own `.adoc` file in the `docs` folder, e.g. `docs/report-entries.adoc`
* [x] Site documentation in `.adoc` file references demo in `src/demo/java` instead of containing code blocks as text
* [x] Only one sentence per line (especially in `.adoc` files)
* [x] Javadoc uses formal style, while sites documentation may use informal style

Documentation (new extension)
* [x] The `docs/docs-nav.yml` navigation has an entry for the new extension
* [x] The `package-info.java` contains information about the new extension

Code
* [x] Code adheres to code style, naming conventions etc.
* [x] Successful tests cover all changes
* [x] There are checks which validate correct / false usage / configuration of a functionality and there are tests to verify those checks
* [x] Tests use [AssertJ](https://assertj.github.io/doc/) or our own [PioneerAssert](https://github.com/junit-pioneer/junit-pioneer/blob/main/CONTRIBUTING.md#assertions) (which are based on AssertJ)

Contributing
* [x] A prepared commit message exists
* [x] The list of contributions inside `README.md` mentions the new contribution (real name optional)

### CODE REVIEW

**Comment by nipafx** [comment]
@jbduncan I made a bunch of changes to the resource and temp dir documentations. Would be nice if you had the time to look them over. 😃

**Comment by Bukama** [comment]
I didn't want to open that topic, bus as you did: Did you consider working on accessibility? 🙈

**Comment by nipafx** [comment]
I just ran two tools and the website is looking ok. Since there are serious considerations to replace it with something else (see junit-pioneer/junit-pioneer.github.io#26), I would not spend much time on it now.

File: docs/cartesian-product-v1.adoc [hunk 1]
```diff
@@ -1,11 +1,14 @@
-:page-title: Old Cartesian product of all test parameters (deprecated in 1.6.0)
-:page-description: Extends JUnit Jupiter with `@CartesianProductTest`, a variant of parameterized tests that tests all combinations of its input
+:page-title: Cartesian Product of Parameters [removed in 2.0]
+:page-description: The JUnit 5 (Jupiter) extension `@CartesianTest` adds a variant of parameterized tests that executes a test for each possible combination of input arguments
```
Comment (Michael1993) [inline docs/cartesian-product-v1.adoc:2]: All mentions of `CartesianTest` should be `CartesianProductTest` in this document.

**Comment by Bukama** [comment]
> 

it looks okay except the hundres of contrast issue, from the code examples (as we also use asciidoc we have to search for a correct code style renderer [which is not task of asciidoc]), the layout tables and so on. But yeah we can tackle this in a different issue and focus here on content.

File: docs/temp-directory.adoc [hunk 2]
```diff
@@ -1,20 +1,23 @@
...
-:page-description: Extends JUnit Jupiter with a mechanism to create, share, and inject temporary directories.
+:page-title: Injecting Temporary Directories
+:page-description: The JUnit 5 (Jupiter) extension `@Dir` allows to create, share, and inject temporary directories as arguments
 :xp-demo-dir: ../src/demo/java
 :demo: {xp-demo-dir}/org/junitpioneer/jupiter/resource/ResourceExtensionDemo.java
 :resources-doc: ./resources.adoc
 
+A temporary directory is a directory on the machine's on-disk filesystem that is created for one or more tests and deleted when it is no longer needed.
+Temporary directories are created via annotations on parameters and configure whether a directory is created anew for a test or shared with others.
+Either way, they are deleted once they re no longer needed.
```
Comment (jbduncan) [inline docs/temp-directory.adoc:9]: Minor typo: "re" should be replaced with "are".

File: docs/resources.adoc [hunk 3]
```diff
@@ -126,12 +124,20 @@ You can create as many global shared resources as you want, as long as they have
...
+== Cleaning Up Resources
+
+Resources will be cleaned up, meaning `close` will be called on them and they will be made eligible for garbage collection, when they are no longer needed:
+
+* for a `@New` parameter, this happens immediately after the test
+* for a `@Shared` parameter with scope `SOURCE_FILE`, this happens when all tests in that source file were executed
+* for a `@Shared` parameter with scope `GLOBAL`, this happens when the entire test suite was executed
+
+Resources will be cached in-memory until then.
+
```
Comment (jbduncan) [inline docs/resources.adoc:140]: I like this, especially how to clarifies what "cleaning up" a resource means, given that the documentation uses both the words "clean" and "close". :)

File: docs/resources.adoc [hunk 4]
```diff
@@ -361,4 +364,5 @@ Temporary directories are a good example of this, as tests can create new subdir
 [CAUTION]
 ====
 Be careful not to save resources in fields from any test method, including `@BeforeAll` and `@BeforeEach` methods, as this extension cannot guarantee that such resources are read or mutated sequentially.
+Instead, use `@Shared` to reuse resources in multiple tests.
```
Comment (jbduncan) [inline docs/resources.adoc:367]: Great catch!

File: src/main/java/org/junitpioneer/jupiter/resource/ResourceExtension.java [hunk 5]
```diff
@@ -431,6 +431,10 @@ private ExtensionContext.Store scopedStore(ExtensionContext extensionContext, Sh
 
 	private ExtensionContext scopedContext(ExtensionContext extensionContext, Shared.Scope scope) {
 		if (scope == Shared.Scope.SOURCE_FILE) {
+			// search for the test scope that's associated with the same source file,
+			// which we assume is the one that has the root context as parent
+			// (contexts in between the test method context and the source class context
+			//  would belong to nested test classes)
```
Comment (jbduncan) [inline src/main/java/org/junitpioneer/jupiter/resource/ResourceExtension.java:437]: Nice, clarifying comment!

File: src/demo/java/org/junitpioneer/jupiter/params/RangeSourcesDemo.java [hunk 6]
```diff
@@ -53,6 +45,14 @@ void illegalRange(byte arg) {
 			// since the range will be empty
 		}
 
+		@ParameterizedTest
+		@LongRangeSource(from = 0L, to = 0L, closed = true)
+		void legalRrange(long arg) {
```
Comment (jbduncan) [inline src/demo/java/org/junitpioneer/jupiter/params/RangeSourcesDemo.java:50]: I wonder if there is a typo here. Should it be spelled `legalRrange` with the extra lowercase "r", or without it?

**Review by jbduncan — CHANGES_REQUESTED** [review]
Great improvements, @nipafx! Just a small number of typos and clarifying questions, then I'd be totally happy.

File: src/demo/java/org/junitpioneer/jupiter/params/RangeSourcesDemo.java (hunk 6 above)
Comment (nipafx) [reply src/demo/java/org/junitpioneer/jupiter/params/RangeSourcesDemo.java]: Lol, good catch back to you!

**Comment by nipafx** [comment]
@jbduncan I added tests to verify the correct cleanup behavior in b11815a.

File: docs/cartesian-product-v1.adoc (hunk 1 above)
Comment (nipafx) [reply docs/cartesian-product-v1.adoc]: Good catch, stupid copy-paste error.

**Comment by nipafx** [comment]
As I see it, this is ready to merge. Let's run the full build and if nobody has any concerns, a maintainer can merge it.

**Comment by jbduncan** [comment]
@nipafx LGTM!

**Pull request merged**